### ISSUE TITLE
klogg does not reflect changes in log file unless glogg is open

### ISSUE THREAD

**Issue by safield** [variar/klogg#45]
I am running on Windows 10 1809.

I just installed Klogg version 19.1.0.332.

Previous to that, I already had glogg v1.1.4-x86_64 installed.

When I load a log file in klogg that is actively being written to, it does not show any newly written lines, after the lines that were shown when the file was first loaded. Essentially it is static.

If I then open **g**logg, while keeping open **k**logg, **k**logg will now correctly be updating the new lines being written to the file.

**Comment by variar** [comment]
Could you run klogg with `klogg -dd -l` command? That should produce a log file in the same directory where executable is (need to copy klogg from Program Files first, or use portable version).

Is option to keep file closed enabled?

What program is used for file writing? I test this by editing file in Notepad++. It might be that Notepad++ is  influencing file changes notifications the same way glogg does so my test is incorrect.

**Comment by variar** [comment]
Did some testing. On Windows FILE_NOTIFY_CHANGE_LAST_WRITE and FILE_NOTIFY_CHANGE_SIZE notifications are sent only when the file is written to the disk. From MSDN:
> For operating systems that use extensive caching, detection occurs only when the cache is sufficiently flushed

Looks like cache is flushed when file is opened by some program. 

I've created a test tool that writes small strings to file once in several seconds. It does not flush the cache so changes do not appear in klogg. When I open this file in notepad, changes appear in klogg, but only once. 

According to process monitor logs glogg is opening file to check its modification time every four seconds. That causes file cache to be flushed and then klogg detects changes. 

I'll add something like this periodic check to force cache flush.

**Comment by variar** [comment]
Added back polling feature from glogg. It can be enabled in settings. Changes to this settings are applied on any file load/close operations. It will not affect currently loaded files until some file is reloaded manually or closed.

**Comment by variar** [comment]
> Tested klogg-19.01.0.337-x64-portable.zip : 'Enable Polling' doesn't seem to work. :(
> I'm monitoring a log on a network share updated almost every second. Klogg doesn't detect any update to the log after it has been opened.

_Originally posted by @wiz0u in https://github.com/variar/klogg/issues/36#issuecomment-459797246_

Neen investigation.  Looks like file monitoring api works with network drives, but only if the remote server supports the functionality

**Comment by wiz0u** [comment]
(Are you sure this issue is adequate for network shares issues? Maybe I should create a separate issue?)

Ok now I'm really confused. Until recently it seems Glogg 1.0.2 would work correctly on network shares, but now it doesn't pick up log updates unless Glogg 1.1.4 is also opened & monitoring the file.
As for klogg 1.0.337, it doesn't pick up any log update at all. (with Enable polling on or off)
I wonder if something was changed on the NAS server (it's handled by another team, i don't know the config)

**Comment by variar** [comment]
> Are you sure this issue is adequate for network shares issues? Maybe I should create a separate issue?

If polling works on local files, we should move issue with network shares to a separate bug. As I've said, I don't use live file updates so can't really tell if the feature is working in real life scenarios

**Comment by variar** [comment]
In my case klogg can't add network share to watch list because it receives `Access Denied` error. This is seen in logs (need to use portable version and start with `-d -l` options to get log file in the same folder where executable is)
```
13:44:22.095 WARN  [6848] [EfswFileWatcher::addFile@75] QtFileWatcher::addFile failed to add watch //example.com/global/nas/Filimonov error -5
```

**Comment by variar** [comment]
Please try build 19.02.0.343.

p.s.
At the moment changes of polling settings are not applied to already opened files, need to reload one of them or open some new file.

**Comment by wiz0u** [comment]
Just tested 19.02.0.343 with "Keep file closed" checked
(as expected, if "Keep file closed" is unchecked, I couldn't delete the log file)
Remote files: Works with Enable Polling on. Doesn't work with Enable Polling off.
Local files: Works fine in both case

Seems great like this :)

**Comment by variar** [comment]
Thanks. Will close this one.
